application/core/lookup_functions.py

```diff
@@ -0,0 +1,223 @@
+    custom_temp_dir = "./var"
+
+    # normalise phase inputs
+    skip_patterns = pipeline.skip_patterns(resource)
+    null_path = None
+
+    # concatfieldphase
+    concats = pipeline.concatenations(resource)
+    column_field_log = ColumnFieldLog(dataset=dataset, resource=resource)
+
+    # map phase
+    intermediate_fieldnames = specification.intermediate_fieldnames(pipeline)
+    columns = pipeline.columns(resource)
+
+    # patch phase
+    patches = pipeline.patches(resource=resource)
+
+    # harmonize phase
+    issue_log = IssueLog(dataset=dataset, resource=resource)
+
+    # default phase
+    default_fields = pipeline.default_fields(resource=resource)
+    default_values = pipeline.default_values(endpoints=[])
+
+    if len(organisations) == 1:
+        default_values["organisation"] = organisations[0]
+
+    # migrate phase
+    schema = specification.pipeline[pipeline.name]["schema"]
+
+    # organisation phase
+    organisation = Organisation(
+        os.path.join("var", "cache", "organisation.csv"), Path(pipeline.path)
+    )
+
+    # print lookups phase
+    flookups = pipeline.lookups()
+
+    run_pipeline(
+        ConvertPhase(
+            path=input_path,
+            dataset_resource_log=dataset_resource_log,
+            custom_temp_dir=custom_temp_dir,
+        ),
+        NormalisePhase(skip_patterns=skip_patterns, null_path=null_path),
+        ParsePhase(),
+        ConcatFieldPhase(concats=concats, log=column_field_log),
+        MapPhase(
+            fieldnames=intermediate_fieldnames,
+            columns=columns,
+            log=column_field_log,
+        ),
+        FilterPhase(filters=pipeline.filters(resource)),
+        PatchPhase(
+            issues=issue_log,
+            patches=patches,
+        ),
+        HarmonisePhase(
+            specification=specification,
+            issues=issue_log,
+        ),
+        DefaultPhase(
+            default_fields=default_fields,
+            default_values=default_values,
+            issues=issue_log,
+        ),
+        # TBD: move migrating columns to fields to be immediately after map
+        # this will simplify harmonisation and remove intermediate_fieldnames
+        # but effects brownfield-land and other pipelines which operate on columns
+        MigratePhase(
+            fields=specification.schema_field[schema],
+            migrations=pipeline.migrations(),
+        ),
+        OrganisationPhase(
+            organisation=organisation,
+            issues=issue_log,
+        ),
+        FieldPrunePhase(fields=specification.current_fieldnames(schema)),
+        EntityReferencePhase(
+            dataset=dataset,
+            specification=specification,
+        ),
+        EntityPrefixPhase(dataset=dataset),
+        UnassignedEntries(lookups=flookups),
+        SavePhase(
+            path=Path("./var/cache/unassigned-entries.csv"),
+            fieldnames=["prefix", "organisation", "reference"],
+        ),
+    )
+
+
```

Comment:
There are methods to do this in digital-land-python now rather than just the code I wrote in the endpoint checker, have a look at add_endpoint_and_lookups_cmd indigital-land-python. This probably doesn't need sorting now but probably should be done.

Reply:
add_endpoint_and_lookups_cmd method in digital-land-python